main.go

```diff
@@ -64,28 +61,22 @@ func main() {
 	go serveMetrics(cfg.MetricsAddress)
 	go handleSigterm(stopChan)
 
-	client, err := newClient(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	serviceSource, err := source.NewServiceSource(client, cfg.Namespace, cfg.FqdnTemplate, cfg.Compatibility)
-	if err != nil {
-		log.Fatal(err)
-	}
-	source.Register("service", serviceSource)
-
-	ingressSource, err := source.NewIngressSource(client, cfg.Namespace, cfg.FqdnTemplate)
-	if err != nil {
-		log.Fatal(err)
+	// Create a source.Config from the flags passed by the user.
+	sourceCfg := source.Config{
+		KubeMaster:    cfg.Master,
+		KubeConfig:    cfg.KubeConfig,
+		Namespace:     cfg.Namespace,
+		FQDNTemplate:  cfg.FQDNTemplate,
+		Compatibility: cfg.Compatibility,
 	}
-	source.Register("ingress", ingressSource)
 
-	sources, err := source.LookupMultiple(cfg.Sources)
+	// Lookup all the selected sources and pass them the desired configuration.
+	sources, err := source.LookupMultiple(cfg.Sources, &sourceCfg)
```

Comment:
Just discovered an issue: We pass a `sourceCfg` pointer but I remember modifying the value inside the constructor functions that are called. This could lead to funny outcomes.

I can image two ways to solve it:
* Don't pass pointers but copies.
* Have a contract with any constructor function not to modify the incoming parameters.

Not sure what's more go-ish?